src/neo4j_genai/experimental/components/pdf_loader.py

```diff
@@ -26,17 +28,30 @@
 from neo4j_genai.experimental.pipeline import Component, DataModel
 
 
+class DocumentInfo(DataModel):
+    path: str
+    metadata: Optional[Dict[str, str]] = None
```

Comment:
This is a bit of a meta-point not related directly to this review but are all of the small Pydantic classes like this really needed? We have a lot of them in the package now and they seem a bit unnecessary in some instances, especially when they're just internal to the package and aren't used for validating user input. I think type hints would be enough in those scenarios

Reply:
In this specific case, we use it to validate the parameter propagation defined in the `Pipeline.connect` method. But in general this is true, we can rethink how we do this.